src/Commands/RunCommand.cs

```diff
@@ -28,4 +32,57 @@ public override string GetCommandName()
 
     public string ScriptToRun { get; set; }
     public ScriptType Type { get; set; }
+    public int TimeoutMilliseconds { get; set; }
+
+    public async Task<int> ExecuteWithTimeoutAsync(Process process)
```

Comment:
We don't need this method. Everything else you did works great, just remove this.